snooty/test_parser.py

```diff
@@ -612,6 +612,92 @@ def test_accidental_indentation() -> None:
     assert len(diagnostics) == 1
 
 
+def test_glossary_node() -> None:
+    path = ROOT_PATH.joinpath(Path("test.rst"))
+    project_config = ProjectConfig(ROOT_PATH, "", source="./")
+    parser = rstparser.Parser(project_config, JSONVisitor)
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+.. glossary::
+  :sorted:
+
+  _id
+    foofoobarbar
+
+  index 
+    foofoofoofoobarbarbarbar
+
+  $cmd
+    foobar
+    
+  aggregate
+    foofoofoobarbarbar
+  
+  
+""",
+    )
+    page.finish(diagnostics)
+
+    check_ast_testing_string(
+        page.ast,
+        """
+<root> 
+  <directive name="glossary" sorted="True">
+    <definitionList>       
+      <definitionListItem>         
+        <term>           
+          <text>$cmd</text>
+          <inline_target domain="std" name="term">            
+            <target_identifier ids="['cmd']">
+              <text>$cmd</text></target_identifier>           
+          </inline_target>         
+        </term>
+        <paragraph><text>foobar</text></paragraph>       
+      </definitionListItem>        
+
+      <definitionListItem>
+        <term>           
+          <text>_id</text>           
+          <inline_target domain="std" name="term">
+            <target_identifier ids="['id']">               
```

Comment:
why the target id is not _id: "For CSS compatibility, identifiers (the “class” and “id” attributes) should have no underscores, colons, or periods. Hyphens may be used." ([docutils docs])(http://code.nabla.net/doc/docutils/api/docutils/nodes/docutils.nodes.make_id.html)

Reply:
:+1:
(Note that while we currently use `make_id()`, I don't consider that a long-term solution, and I'll re-evaluate in DOP-998)